File: src/main/java/org/jukeboxmc/network/handler/CommandRequestHandler.java
package org.jukeboxmc.network.handler;

import org.cloudburstmc.protocol.bedrock.packet.CommandRequestPacket;
import org.jukeboxmc.Server;
import org.jukeboxmc.event.player.PlayerCommandPreprocessEvent;
import org.jukeboxmc.player.Player;

/**
 * @author LucGamesYT
 * @version 1.0
 */
public class CommandRequestHandler implements PacketHandler<CommandRequestPacket>{

    @Override
    public void handle( CommandRequestPacket packet, Server server, Player player ) {
        final String[] commandParts = packet.getCommand().substring( 1 ).split( " " );
        final String commandIdentifier = commandParts[0];
        PlayerCommandPreprocessEvent playerCommandPreprocessEvent = new PlayerCommandPreprocessEvent( player, commandIdentifier );
        server.getPluginManager().callEvent( playerCommandPreprocessEvent );

        if ( playerCommandPreprocessEvent.isCancelled() ) {
            return;
        }
        server.getPluginManager().getCommandManager().handleCommandInput( player, packet.getCommand().toLowerCase() );
    }
}

File: JukeboxMC-Server/src/main/kotlin/org/jukeboxmc/server/network/handler/CommandRequestHandler.kt
package org.jukeboxmc.server.network.handler

import org.cloudburstmc.protocol.bedrock.packet.CommandRequestPacket
import org.jukeboxmc.api.event.player.PlayerCommandPreprocessEvent
import org.jukeboxmc.server.JukeboxServer
import org.jukeboxmc.server.player.JukeboxPlayer

class CommandRequestHandler : PacketHandler<CommandRequestPacket> {

    override fun handle(packet: CommandRequestPacket, server: JukeboxServer, player: JukeboxPlayer) {
        val commandIdentifier = packet.command.substring(1).split(" ").toTypedArray()[0]
        val playerCommandProcessEvent = PlayerCommandPreprocessEvent(player, commandIdentifier)
        server.getPluginManager().callEvent(playerCommandProcessEvent)
        if (playerCommandProcessEvent.isCancelled()) return
        server.getCommandManager().handleCommandInput(player, packet.command)
    }
}